jena-text/src/main/java/org/apache/jena/query/text/TextQueryPF.java

```diff
@@ -347,11 +352,15 @@ private StrMatch objectToStruct(PropFuncArg argObject, boolean executionTime) {
                 log.warn("Text query string is not a literal " + list) ;
             return null ;
         }
-        
-        if (x.getLiteralDatatype() != null && !x.getLiteralDatatype().equals(XSDDatatype.XSDstring)) {
```

Comment:
Comment:
In RDF 1.1, the data type of a literal is never null.

There are functions like `Util.isLangString` (in RDF 1.1: rdf:langString) and `Util.isSimpleString` (in RDF 1.1, xsd:string) that mask the RDF1.0/RDF 1.1 mode setting (not that Jena claims to support RDF 1.0 any more).

Reply:
Thank you for the tips. I tried to avoid changing code that was present. I plan to raise another issue or two in jena-text and will be glad to visit some of these cases to try and use RDF functionality rather than bare Java functionality. It certainly will make the code clearer.